src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -749,42 +746,14 @@ def _flatten_kwargs(self, kwargs, default_source_name):
 
     # pylint: disable=arguments-differ
     def command(self, name, method_name=None, **kwargs):
-        """
-        Register a CLI command
-        :param name: Name of the command as it will be called on the command line
-        :type name: str
-        :param method_name: Name of the method the command maps to
-        :type method_name: str
-        :param kwargs: Keyword arguments. Supported keyword arguments include:
-            - client_factory: Callable which returns a client needed to access the underlying command method. (function)
-            - confirmation: Prompt prior to the action being executed. This is useful if the action
-                            would cause a loss of data. (bool)
-            - exception_handler: Exception handler for handling non-standard exceptions (function)
-            - supports_no_wait: The command supports no wait. (bool)
-            - no_wait_param: [deprecated] The name of a boolean parameter that will be exposed as `--no-wait`
-              to skip long-running operation polling. (string)
-            - transform: Transform function for transforming the output of the command (function)
-            - table_transformer: Transform function or JMESPath query to be applied to table output to create a
-                                 better output format for tables. (function or string)
-            - resource_type: The ResourceType enum value to use with min or max API. (ResourceType)
-            - min_api: Minimum API version required for commands within the group (string)
-            - max_api: Maximum API version required for commands within the group (string)
-        :rtype: None
-        """
-        self._check_stale()
-        merged_kwargs = self._flatten_kwargs(kwargs, 'command_type')
-        # don't inherit deprecation info from command group
-        merged_kwargs['deprecate_info'] = kwargs.get('deprecate_info', None)
-        operations_tmpl = merged_kwargs['operations_tmpl']
-        command_name = '{} {}'.format(self.group_name, name) if self.group_name else name
-        operation = operations_tmpl.format(method_name) if operations_tmpl else None
-        self.command_loader._cli_command(command_name, operation, **merged_kwargs)  # pylint: disable=protected-access
+        return self._command(name, method_name=method_name, **kwargs)
 
-        return command_name
+    def custom_command(self, name, method_name=None, **kwargs):
+        return self._command(name, method_name=method_name, custom_command=True, **kwargs)
```

Comment:
Can confirm that intellisense does not work(at least in vscode), but for what cases would we try to ensure intellisense is enabled?
We have many publicly exposed methods in our CLI, with very few containing docstrings, `command`/`custom_command` were the very rare exceptions. 

IMO, intellisense does not seem very important unless we expect people to use the CLI as a library to call methods from. 
As there is no harm in them, I can add docstring documentation for the command registration methods if you feel strongly otherwise.

Reply:
TL/DR: intellisense is important and we should have the docstrings on those methods

We should have docstrings on any public methods we expect command authors to use. So the public methods of command group and argument context should have docstrings to provide intellisense. A classic example is the `resource_id` method that now lives in msrestazure.tools. This originally was strongly typed, which provided some kind of intellisense. When Johan converted it to just use kwargs, it became impossible to use without pulling up code somewhere to know what you can use. A docstring was added so that the user gets intellisense. This is just as important for these methods that rely heavily on kwargs.